### PR TITLE
Some proposed improvements to error messages

### PR DESCRIPTION
**Pull request by lukechurch**
PTAL: @kronz @lillismith

I couldn't walk away from today and go to bed. Sorry. I had to try to fix some of our errors first. There was something wrong on the internet.

The perfect is the enemy of the good - I left a lot of stuff that would take time to make better alone, and there are a bunch of errors that I'd need to look at how they could be caused to see if they're internal errors or not, I left them alone as well.

Comments, suggested improvements highly encouraged - at this stage lets keep it to things where we only change the words, we don't add or remove new information.

### CODE REVIEW

File: src/Engine/ProtoCore/Properties/Resources.resx [hunk 1]
```diff
@@ -235,22 +235,22 @@
     <value>'{0}()' has some invalid arguments.</value>
   </data>
   <data name="kMethodIsInaccessible" xml:space="preserve">
-    <value>Method '{0}()' is inaccessible.</value>
+    <value>Method '{0}()' is can't be reached from here.</value>
```
Comment (lukechurch) [inline src/Engine/ProtoCore/Properties/Resources.resx]: s/is /

File: src/Engine/ProtoCore/Properties/Resources.resx [hunk 2]
```diff
@@ -259,37 +259,37 @@
...
-    <value>The step number of range expression should be greater than 0.</value>
+    <value>The step number must be greater than 0.</value>
   </data>
   <data name="kRangeExpressionWithNonIntegerStepNumber" xml:space="preserve">
-    <value>The step number of range expression should be integer.</value>
+    <value>The step number must be an integer.</value>
   </data>
   <data name="kRangeExpressionWithStepSizeZero" xml:space="preserve">
-    <value>The step size of range expression should not be 0.</value>
+    <value>The step size of must not be 0.</value>
```
Comment (lukechurch) [inline src/Engine/ProtoCore/Properties/Resources.resx]: remove 'of '

Comment (pboyer) [inline src/Engine/ProtoCore/Properties/Resources.resx]: Are you reminding yourself to fix this?

Comment (lukechurch) [inline src/Engine/ProtoCore/Properties/Resources.resx]: Well, it's kinda lonely out here, so I've taken to talking to myself on code reviews.

But yeah - it's much easier to reread the changes here, catch a few issues, bundle them all up into a single patch than to do the whole change, push palaver to remove an 'of '.

File: src/Engine/ProtoCore/Properties/Resources.resx [hunk 3]
```diff
@@ -160,37 +160,37 @@
     <value>Converting an array to {0} would cause array rank reduction and is not permitted.</value>
   </data>
   <data name="kConvertDoubleToInt" xml:space="preserve">
-    <value>Converting double to int will cause possible information loss.</value>
+    <value>Converting a double to an int may lose information.</value>
   </data>
   <data name="kConvertNonConvertibleTypes" xml:space="preserve">
-    <value>Asked to convert non-convertible types.</value>
+    <value>Asked to convert between non-convertible types.</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.resx]: The object is being converted to non-convertible types, which will result null value.

File: src/Engine/ProtoCore/Properties/Resources.resx [hunk 4]
```diff
@@ -160,37 +160,37 @@
...
-    <value>Failed to obtain object '{0}' for '{1}.{2}'.</value>
+    <value>Internal error, please report: Failed to obtain object '{0}' for '{1}.{2}' (edb6d7d9).</value>
   </data>
   <data name="kFFIFailedToObtainThisObject" xml:space="preserve">
-    <value>Failed to obtain this object for '{0}.{1}'.</value>
+    <value>Internal error, please report: Failed to obtain this object for '{0}.{1}' (88facceb).</value>
   </data>
   <data name="kFFIInvalidCast" xml:space="preserve">
-    <value>'{0}' is being cast to '{1}', but the allowed range is [{2}..{3}].</value>
+    <value>Internal error, please report: '{0}' is being cast to '{1}', but the allowed range is [{2}..{3}]. (dff21092)</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.resx]: This is just a warning message when input type at C# side is byte or short and it's being assigned with higher integer value.

**Comment by kronz** [comment]
@lukechurch LGTM

**Comment by pboyer** [comment]
@lukechurch PTAL This is now insanely old.

**Comment by lukechurch** [comment]
This is obsolete

**Pull request closed without merging**